### DISCUSSION TITLE
Ws2812 instead of blinking led

### DISCUSSION THREAD

**Discussion by Arashinof** [HomeSpan/HomeSpan#143] [Ideas]
I think it would be so cool if homepsan shows status with colors by a ws2812 single chip instead of blinking

**Comment by KhazanGu** [comment]
Plus one.

**Comment by HomeSpan** [comment]
Would a hook into the code so you can invoke your own LED routines be useful?  For example, HomeSpan has default blinking patterns for various types of alerts (WIFI begin acquired, Device Unpaired...).  In principle I could create a callback so that instead of HomeSpan loading up a certain blinking pattern, it calls a function you select.  In that function you could then write your own code to take any action, including setting a ws2812.

**Reply by SnowballsChance** [reply]
That is a good idea. Then the indicator could be almost anything not just an LED but also a buzzer or even a HTTP request.

**Comment by HomeSpan** [comment]
Note the reason HomeSpan does not default to the ws2812 is that it is not a common component on ESP32 boards, whereas most boards have a built-in LED.

**Comment by spacecar** [comment]
A hook would certainly be useful. I have made several devices using boards with tiny displays added. These boards often(?) lack user controllable LEDs because (I assume) they have displays. If I got a callback for each setup state, then it would be simple to prompt the user with a short text as well as hold back the main interface until setup is done.
